test.lisp

```diff
@@ -161,6 +169,29 @@
                               ,@body))
            ',name)))))
 
+(defmacro define-test+run (name &body args-and-body)
+  "Pass NAME with ARGS-AND-BODY to `define-test' and, if the form is executed
+(as opposed to being loaded or compiled), run the test and return a summary of
+the results: if there are failures, return the status, the number of failures
+and the first failed expression; otherwise, return just the status and the
+report.  Useful for interactivity: define and run the test at once, see the
+place that needs fixed right away.  Compilation errors are not muffled."
+  `(progn
+     (define-test ,name ,@args-and-body)
+     (eval-when (:execute)
+       (flet ((failed-p (x) (equal :FAILED (status x))))
+         (let* ((*silence-compilation-errors-p* nil)
+                (report (test ',name :report 'plain))
+                (results (results report))
+                (status (status report)))
+           (cond ((member status '(:PASSED :SKIPPED))
+                  (list status report))
```

Comment:
Returning multiple values instead of a list seems more appropriate here. The first value should still always be the test's name, too.